probe-rs-target/src/chip.rs

```diff
@@ -28,3 +32,18 @@ pub struct Chip {
     /// [`ChipFamily::flash_algorithms`]: crate::ChipFamily::flash_algorithms
     pub flash_algorithms: Vec<String>,
 }
+
+/// An individual core inside a chip
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Core {
+    /// The core name.
+    pub name: String,
+
+    /// The core type.
+    /// E.g. `M0` or `M4`.
+    #[serde(rename = "type")]
+    pub core_type: CoreType,
+
+    /// The AP number to access the core
+    pub arm_swd_ap: u8,
```

Comment:
We should get rid of the `swd` portion here as the same API is available through JTAG too if JTAG is available.

Reply:
Done